src/lib.rs

```diff
@@ -516,11 +520,20 @@ macro_rules! quote_bind_into_iter {
 //       };
 //
 // in:   quote_bind_next_or_break!({})
-// out:  break;
+// out:  if true {
+//           break;
+//       }
 #[macro_export(local_inner_macros)]
 #[doc(hidden)]
 macro_rules! quote_bind_next_or_break {
-    ({}) => { break; };
+    ({}) => {
+        // The code will fail to compile because there are no iterators used
+        // inside of this repetition. The following is just here to bypass some
+        // unreachable statement warnings.
```

Comment:
Are you thinking of one of the following?

```diff
- if true {
-     break;
- }
+ loop {}
```

```diff
  if true {
-     break;
+     loop {}
  }
```

```diff
- if true {
-     break;
- }
```

When I tried it, all three of those produce an extra unreachable code warning in addition to the error about HasIter being false, which I am trying to avoid.

Reply:
Nah, I was imagining just generating no code at all, but I suppose that might produce an unreachable code warning for the code after the loop?

I suppose we could also make the outer `loop` construct do something like `while has_iter.reify() { ... }` where `reify` always returns either `true` or `false`?